### ISSUE TITLE
should be using absolute URLs according to the spec

### ISSUE THREAD

**Issue by michielbdejong** [solid/node-solid-ws#8]
https://github.com/solid/solid-spec/blob/master/api-websockets.md
So `sub https://localhost:8443/public/` and not `sub /public/` or `sub public`

**Comment by RubenVerborgh** [comment]
Agreed, but a warning for implementers: the module might not know the external URL (might be behind a proxy etc.), which is part of (for instance) NSS config.

**Comment by michielbdejong** [comment]
Yes, in that case it should look at the `X-Forwarded-Host` header, I think? The LDP server probably also wants to know which absolute URLs it can consider as local.

**Comment by RubenVerborgh** [comment]
Indeed, that one and related headers (https://tools.ietf.org/html/rfc7239).
Also theoretically need to distinguish between HTTP and HTTPS, but we should be able to safely assume the latter (although you never know in test environments).

Trouble is that NSS will prefer its configured URL and the Host header over these, but that's probably a NSS problem then.
